x/rollapp/keeper/fraud_handler.go

```diff
@@ -42,34 +37,22 @@ func (k Keeper) HandleFraud(ctx sdk.Context, rollappID, clientId string, fraudHe
 		return errorsmod.Wrapf(types.ErrWrongProposerAddr, "sequencer address %s does not match the one in the state info", seqAddr)
 	}
 
+	// check that the clientID is correct
+	err = k.verifyClientID(ctx, rollappID, clientID)
+	if err != nil {
+		return errors.Join(types.ErrWrongClientId, err)
+	}
+
 	// slash the sequencer, clean delayed packets
 	err = k.hooks.FraudSubmitted(ctx, rollappID, fraudHeight, seqAddr)
 	if err != nil {
 		return err
 	}
 
-	// mark the rollapp as frozen. revert all pending states to finalized
-	rollapp.Frozen = true
-	k.SetRollapp(ctx, rollapp)
-
-	k.RevertPendingStates(ctx, rollappID)
-
-	if rollapp.ChannelId != "" {
-		extractedClientId, _, err := k.channelKeeper.GetChannelClientState(ctx, "transfer", rollapp.ChannelId)
-		if err != nil {
-			return err
-		}
-
-		if extractedClientId != clientId {
-			return errorsmod.Wrapf(types.ErrWrongClientId, "clientID %s does not match the one in the rollapp (%s)", clientId, extractedClientId)
-		}
-
-		err = k.freezeClientState(ctx, clientId)
-		if err != nil {
-			return err
-		}
-	} else if clientId != "" {
-		return errorsmod.Wrapf(types.ErrWrongClientId, "rollapp with ID %s does not have a channel", rollappID)
+	// freeze the rollapp and revert all pending states
+	err = k.FreezeRollapp(ctx, rollappID)
+	if err != nil {
+		return fmt.Errorf("freeze rollapp: %w", err)
```

Comment:
maybe this should come before the hook call?

Can you also make it clear in a comment that the freeze stuff is subject to change soon?